Ledger 3.2.1 fails to build on Darwin

**Describe the bug**

Since the recent commit [ledger: 3.1.3 -> 3.2.1](c9c8acf) (@jonringer), ledger fails to build on Darwin.

**To Reproduce**

`nix-shell -p ledger` or similar.

**Additional context**
Add any other context about the problem here.

```
building '/nix/store/q3pbx11j8qjd1v4rbxwslmd84vnmysvg-ledger-3.2.1.drv'...
unpacking sources
unpacking source archive /nix/store/vd5vjh5gy6gmbdv5kfq1x99j1g0ddryh-source
source root is source
patching sources
configuring
fixing cmake files...
grep: Invalid range end
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_SKIP_BUILD_RPATH=ON -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/2qcvknjkkhyx0qs13h1jxa2dgj00s95y-ledger-3.2.1/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/2qcvknjkkhyx0qs13h1jxa2dgj00s95y-ledger-3.2.1/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/2qcvknjkkhyx0qs13h1jxa2dgj00s95y-ledger-3.2.1/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/2qcvknjkkhyx0qs13h1jxa2dgj00s95y-ledger-3.2.1/share/doc/ledger -DCMAKE_INSTALL_INFODIR=/nix/store/2qcvknjkkhyx0qs13h1jxa2dgj00s95y-ledger-3.2.1/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/2qcvknjkkhyx0qs13h1jxa2dgj00s95y-ledger-3.2.1/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/1jpcb89rdp3d48rb09krkpml76xsfkfg-ledger-3.2.1-dev/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/1jpcb89rdp3d48rb09krkpml76xsfkfg-ledger-3.2.1-dev/include -DCMAKE_INSTALL_SBINDIR=/nix/store/2qcvknjkkhyx0qs13h1jxa2dgj00s95y-ledger-3.2.1/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/2qcvknjkkhyx0qs13h1jxa2dgj00s95y-ledger-3.2.1/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/2qcvknjkkhyx0qs13h1jxa2dgj00s95y-ledger-3.2.1/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_SYSROOT= -DCMAKE_OSX_ARCHITECTURES=x86_64 -DCMAKE_FIND_FRAMEWORK=last -DCMAKE_STRIP=/nix/store/5vwyr9w0cy0g7hlzfrpnqas495av3xhj-cctools-binutils-darwin-927.0.2/bin/strip -DCMAKE_RANLIB=/nix/store/5vwyr9w0cy0g7hlzfrpnqas495av3xhj-cctools-binutils-darwin-927.0.2/bin/ranlib -DCMAKE_AR=/nix/store/5vwyr9w0cy0g7hlzfrpnqas495av3xhj-cctools-binutils-darwin-927.0.2/bin/ar -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_INSTALL_PREFIX=/nix/store/2qcvknjkkhyx0qs13h1jxa2dgj00s95y-ledger-3.2.1 -DCMAKE_INSTALL_LIBDIR=lib -DBUILD_DOCS:BOOL=ON -DUSE_PYTHON=true
-- The C compiler identification is Clang 7.1.0
-- The CXX compiler identification is Clang 7.1.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/0nl15nlwb8c6bz38w87hwfqs2bphwvaj-clang-wrapper-7.1.0/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/0nl15nlwb8c6bz38w87hwfqs2bphwvaj-clang-wrapper-7.1.0/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Warning (dev) at /nix/store/039xvnxlvkz45miz3p8yvm3kj2jbkp84-cmake-3.18.0/share/cmake-3.18/Modules/FindPython/Support.cmake:1211 (message):
  FindPython: last: invalid value for 'CMAKE_FIND_FRAMEWORK'.  'FIRST',
  'LAST' or 'NEVER' expected.  'FIRST' will be used instead.
Call Stack (most recent call first):
  /nix/store/039xvnxlvkz45miz3p8yvm3kj2jbkp84-cmake-3.18.0/share/cmake-3.18/Modules/FindPython.cmake:415 (include)
  CMakeLists.txt:61 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /nix/store/039xvnxlvkz45miz3p8yvm3kj2jbkp84-cmake-3.18.0/share/cmake-3.18/Modules/FindPython/Support.cmake:1211 (message):
  FindPython: last: invalid value for 'CMAKE_FIND_FRAMEWORK'.  'FIRST',
  'LAST' or 'NEVER' expected.  'FIRST' will be used instead.
Call Stack (most recent call first):
  /nix/store/039xvnxlvkz45miz3p8yvm3kj2jbkp84-cmake-3.18.0/share/cmake-3.18/Modules/FindPython.cmake:415 (include)
  CMakeLists.txt:61 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found Python: /System/Library/Frameworks/Python.framework/Versions/2.7/bin/python2.7 (found version "2.7.16") found components: Interpreter
CMake Warning (dev) at /nix/store/039xvnxlvkz45miz3p8yvm3kj2jbkp84-cmake-3.18.0/share/cmake-3.18/Modules/FindPython/Support.cmake:1211 (message):
  FindPython: last: invalid value for 'CMAKE_FIND_FRAMEWORK'.  'FIRST',
  'LAST' or 'NEVER' expected.  'FIRST' will be used instead.
Call Stack (most recent call first):
  /nix/store/039xvnxlvkz45miz3p8yvm3kj2jbkp84-cmake-3.18.0/share/cmake-3.18/Modules/FindPython.cmake:415 (include)
  CMakeLists.txt:68 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found Python: /System/Library/Frameworks/Python.framework/Versions/2.7/bin/python2.7 (found version "2.7.16") found components: Interpreter Development Development.Module Development.Embed
-- Found Boost: /nix/store/n4splbsv1b1130k2n16mhaz45mspqysn-boost-1.69.0-dev/include (found suitable version "1.69.0", minimum required is "1.49.0") found components: date_time filesystem system iostreams regex unit_test_framework python27
-- Looking for getpwuid
-- Looking for getpwuid - found
-- Looking for getpwnam
-- Looking for getpwnam - found
-- Looking for ioctl
-- Looking for ioctl - found
-- Looking for isatty
-- Looking for isatty - found
-- Performing Test UNIX_PIPES_COMPILES
-- Performing Test UNIX_PIPES_COMPILES - Success
-- Performing Test BOOST_REGEX_UNICODE_RUNS
-- Performing Test BOOST_REGEX_UNICODE_RUNS - Failed
-- Performing Test BOOST_MAKE_SETTER_COMPILES
-- Performing Test BOOST_MAKE_SETTER_COMPILES - Failed
-- Looking for readline in edit
-- Looking for readline in edit - found
-- Performing Test HAVE_WORKING_UTFCPP
-- Performing Test HAVE_WORKING_UTFCPP - Success
CMake Warning at doc/CMakeLists.txt:52 (message):
  Could not find texi2pdf or tex.  PDF version of documentation will not be
  built.


-- Configuring done
CMake Warning (dev):
  Policy CMP0042 is not set: MACOSX_RPATH is enabled by default.  Run "cmake
  --help-policy CMP0042" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  MACOSX_RPATH is not specified for the following targets:

   libledger

This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev):
  Policy CMP0068 is not set: RPATH settings on macOS do not affect
  install_name.  Run "cmake --help-policy CMP0068" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

  For compatibility with older versions of CMake, the install_name fields for
  the following targets are still affected by RPATH settings:

   libledger

This warning is for project developers.  Use -Wno-dev to suppress it.

-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY


-- Build files have been written to: /tmp/nix-build-ledger-3.2.1.drv-0/source/build
building
build flags: -j2 -l2 SHELL=/nix/store/d50mzwc57fgc6j1gf320hcwc81k0jdph-bash-4.4-p23/bin/bash
Scanning dependencies of target libledger
[  1%] Building CXX object src/CMakeFiles/libledger.dir/stats.cc.o
In file included from /tmp/nix-build-ledger-3.2.1.drv-0/source/src/stats.cc:32:
In file included from /tmp/nix-build-ledger-3.2.1.drv-0/source/build/system.hh:207:
In file included from include/boost/python.hpp:11:
In file included from include/boost/python/args.hpp:8:
In file included from include/boost/python/detail/prefix.hpp:13:
include/boost/python/detail/wrap_python.hpp:50:11: fatal error: 'pyconfig.h' file not found
# include <pyconfig.h>
          ^~~~~~~~~~~~
1 error generated.
make[2]: *** [src/CMakeFiles/libledger.dir/build.make:82: src/CMakeFiles/libledger.dir/stats.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:242: src/CMakeFiles/libledger.dir/all] Error 2
make: *** [Makefile:182: all] Error 2
builder for '/nix/store/q3pbx11j8qjd1v4rbxwslmd84vnmysvg-ledger-3.2.1.drv' failed with exit code 2
```

**Notify maintainers**

@jwiegley 

**Metadata**

 - system: `"x86_64-darwin"`
 - host os: `Darwin 19.6.0, macOS 10.15.6`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.7`
 - channels(steve): `"darwin, nixpkgs"`
 - channels(root): `"darwin, master.tar.gz"`
 - nixpkgs: `/Users/steve/.nix-defexpr/channels/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: ledger
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/applications/office/ledger/default.nix
+++ b/pkgs/applications/office/ledger/default.nix
@@ -27,6 +27,10 @@ stdenv.mkDerivation rec {
     "-DCMAKE_INSTALL_LIBDIR=lib"
     "-DBUILD_DOCS:BOOL=ON"
     (lib.optionalString usePython "-DUSE_PYTHON=true")
+  ] ++ lib.optionals (usePython && stdenv.isDarwin) [
+    # Fix python lookup on Darwin. Not necessary after
+    # https://github.com/NixOS/nixpkgs/pull/94090 lands in master
+    "-DPython_ROOT_DIR=${python}"
   ];
 
   # by default, it will query the python interpreter for it's sitepackages location

